File: Program's_Contributed_By_Contributors/C++_Programs/Sorting/BubbleSort.cpp
#include <bits/stdc++.h>

using namespace std;

void bubbleSort(vector<int> v)
{
    int i = 0, t = 0;

    for (int i = 0; i < v.size() - 1; i++)
    {
        for (int j = 0; j < v.size() - i - 1; j++)
        {
            if (v[j] > v[j + 1])
            {
                swap(v[j], v[j + 1]);
                t++;
            }
        }
        if (t == 0)
        {
            cout << "Already sorted" << endl;
        }
    }

    for (int i = 0; i < v.size(); i++)
    {
        cout << v[i] << " ";
    }
}

int main()
{
    int size;
    cin >> size;

    vector<int> v1(size);

    for (auto &i : v1)
    {
        cin >> i;
    }

    bubbleSort(v1);

    return 0;
}

File: Program's_Contributed_By_Contributors/Python_Programs/bubble_sort.py
def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        swapped = False
        for j in range(0, n-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]  # swap the elements
                swapped = True
        # If no swaps occurred inside the loop, the list is sorted
        if not swapped:
            break
    return arr

# Test
arr = [64, 34, 25, 12, 22, 11, 90]
print("Original list:", arr)
bubble_sort(arr)
print("Sorted list:", arr)
